### ISSUE TITLE
Camera LiDAR calibration error is large

### ISSUE THREAD

**Issue by CrisXin000** [tier4/CalibrationTools#101]
Hello, thanks for posting such a great project. I tried to use `extrinsic_interactive_calibrator` for camera-lidar calibration, but the result looks bad, the closer to the edge of the image, the worse the calibration result is.
As shown in below:
![飞书20230509-100022](https://user-images.githubusercontent.com/39207386/237004760-29a96613-df0a-45ab-96d0-c0cbf9b2fad3.jpg)

During the calibration process, I placed three ARUCO markers in the camera field of view, and selected the corner points of the markers as the target points. 
Before calibration:
![Screenshot from 2023-05-09 12-13-08](https://user-images.githubusercontent.com/39207386/237006474-e7cb8149-a34f-484a-b9e1-53ffd225eebd.png)

After calibration:
![Screenshot from 2023-05-09 12-13-30](https://user-images.githubusercontent.com/39207386/237006501-84bb3337-b8aa-46f1-86eb-adba35c4a0e3.png)

I have little experience with sensor calibration, can you give me some help?

**Comment by knzo25** [comment]
Hi @CrisXin000, thanks for checking our tools,

Since you selected points manually, more or less in the center of the image, you can expect that the calibration in that area to be more or less low. However, since you can not choose points perfectly (discrete lidar readings) and we ourselves are a source of error, optimization algorithms would likely fails and overfit to the data, making areas with no points, present a higher-ish reprojection error.

Another possible cause is that your intrinsics are not good enough, in which case there is no possible calibration that gets you a decent reprojection error. You could check that too.

In summary, my recommendations are as follows:
 - Collect points in the whole image. If the result is not good enough it could be either the intrinsics or that your data collection is not good enough.
 - Calibrate intrinsics. We just released an intrinsics calibrator with support for several boards. Sadly, no arucos since we use lidartags.
 - Use the tag-based calibrator. Instead of collecting points manually, you can use lidartags and get them automatically. However, you would need to print at least one of those.

**Comment by CrisXin000** [comment]
Hi @knzo25 ,thanks for replying so quickly,

I understand the source of the calibration error from your answer. I can confirm the intrinsic calibration result is very good (the reprojection error is less than 0.2 pixels), so I think the camera-LiDAR calibration result maybe fall into a local optimum as you said (depending on the manually selected points), I will try placing more markers allows me to extract more image-lidar counterparts.

Also, I would include `tag-based calibrator` as a fallback if I'm not good at manually extracting it.

**Comment by knzo25** [comment]
Glad to be of help !
Please tell us how it goes for you later and remember to close the issue when you feel like it has been resolved :)

**Comment by knzo25** [comment]
@CrisXin000 
As there has been no a activity in this issue, I will be closing it in about one week.
